presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTimeWithTimeZone.java

```diff
@@ -83,6 +117,18 @@ public boolean equals(Object obj)
     @Override
     public String toString()
     {
-        return Instant.ofEpochMilli(millisUtc).atZone(ZoneId.of(timeZoneKey.getId())).format(formatter);
+        // FIXME This is hack that we need to use as the timezone interpretation depends on date (not only on time)
+        // Additionally we cannot just grab current TZ offset from timezoneOffsetReferenceTimestampUtc and use it
+        // as our formatting library will fail to display expected TZ symbol.
+        long currentMillisOfDay =
+                LocalTime.from(
+                        Instant.ofEpochMilli(timezoneOffsetReferenceTimestampUtc)
+                               .atZone(ZoneId.of(TimeZoneKey.UTC_KEY.getId())))
+                        .toNanoOfDay() / 1_000_000L; // nanos to millis
```

Comment:
`timezoneOffsetReferenceTimestampUtc % (86400 * 1000)` - I know that is correct, but this code is hard to understand without weird magic constants.
As for other proposition this would require to to introduce Joda to SPI and as far as we understand we avoid introducing libraries to SPI.